src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1916 @@
+                    }
+
+                    break;
+            }
+
+            // (?)
+            // Parse this as a normal group. The question will immediately error as it's a 
+            // quantifier not following anything.
+            _lexer.Position = afterQuestionPos - 1;
+            return ParseSimpleGroup(openParenToken);
+        }
+
+        private RegexConditionalGroupingNode ParseConditionalGrouping(RegexToken openParenToken, RegexToken questionToken)
+        {
+            var innerOpenParenToken = _currentToken;
+            var afterInnerOpenParen = _lexer.Position;
+
+            var captureToken = _lexer.TryScanNumberOrCaptureName();
+            if (captureToken == null)
+            {
+                return ParseConditionalExpressionGrouping(openParenToken, questionToken, innerOpenParenToken);
+            }
+            else
+            {
+                var capture = captureToken.Value;
+
+                RegexToken innerCloseParenToken;
+                if (capture.Kind == RegexKind.NumberToken)
+                {
+                    // If it's a numeric group, it has to be immediately followed by a )
+                    // and the numeric reference has to exist.
+                    //
+                    // That means that (?(4 ) is not treated as an embedded expression but as
+                    // an error.  This is different from (?(a ) which will be treated as an 
+                    // embedded expression, and different from (?(a) will will be treated as
+                    // an embedded expression or capture group depending on if 'a' is a existing
+                    // capture name.
+
+                    ConsumeCurrentToken(allowTrivia: false);
+                    if (_currentToken.Kind == RegexKind.CloseParenToken)
+                    {
+                        innerCloseParenToken = _currentToken;
+                        if (!HasCapture((int)capture.Value))
+                        {
+                            capture = capture.AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                                WorkspacesResources.Reference_to_undefined_group,
+                                capture.GetSpan()));
+                        }
+                    }
+                    else
+                    {
+                        innerCloseParenToken = CreateMissingToken(RegexKind.CloseParenToken);
+                        capture = capture.AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                            WorkspacesResources.Malformed,
+                            capture.GetSpan()));
+                        MoveBackBeforePreviousScan();
+                    }
+                }
+                else
+                {
+                    // If its a capture name, its ok if it that capture doesn't exist.  In that
+                    // case we will just treat this as an conditional expression.
+                    if (!HasCapture((string)capture.Value))
```

Comment:
How do you know the `Value` will be a `string` here?

Reply:
Will comment.  but captures can only be numbers of strings.  the previous block handles numbers, so this should be safe.